src/geofabrics/bathymetry_estimation.py

```diff
@@ -0,0 +1,1576 @@
+                stop_i = stop_i_list[i]
+            if channel_width >= min_channel_width:
+                channel_count += 1
+
+        return start_i, stop_i, channel_count
+
+    def fixed_threshold_width_outwards(self,
+                                       gnd_samples: numpy.ndarray,
+                                       veg_samples: numpy.ndarray,
+                                       start_index: int,
+                                       z_water: float,
+                                       threshold: float):
+        """ If the start_index is nan or less than the threshold, then cycle
+        outwards until each side has gone above the threshold.
+
+        Parameters
+        ----------
+
+        gnd_samples
+            The ground elevations for a single cross section.
+        veg_samples
+            The vegrtation elevations for the same cross section.
+        start_index
+            The index to start the outward search from.
+        z_water
+            The elevation of the water.
+        threshold
+            The height above the water level to detect as a bank.
+
+        """
+
+        start_i = numpy.nan
+        stop_i = numpy.nan
+
+        if gnd_samples[start_index] - z_water < threshold or \
+                (numpy.isnan(gnd_samples[start_index]) and numpy.isnan(veg_samples[start_index])):
+
+            for i in numpy.arange(0, self.number_of_samples + 1, 1):
+
+                # work forward checking height
+                if start_index + i < self.number_of_samples and numpy.isnan(stop_i):
+                    gnd_elevation_over_minimum = gnd_samples[start_index + i] - z_water
+
+                    # Detect banks - either ground above threshold, or no ground with vegetation over threshold
+                    if numpy.isnan(stop_i) and gnd_elevation_over_minimum > threshold:
+                        # Leaving the channel
+                        stop_i = start_index + i
+
+                # work backward checking height
+                if start_index - i >= 0 and numpy.isnan(start_i):
+                    gnd_elevation_over_minimum = gnd_samples[start_index - i] - z_water
+
+                    # Detect bank
+                    if numpy.isnan(start_i) and gnd_elevation_over_minimum > threshold:
+                        # Leaving the channel
+                        start_i = start_index - i
+
+                # break if both edges detected
+                if not numpy.isnan(start_i) and not numpy.isnan(stop_i):
+                    break
+                # break if both ends of the sampled cross section reached
+                if start_index + i >= self.number_of_samples - 1 and start_index - i <= 0:
+                    if numpy.isnan(start_i):
+                        start_i = 0
+                    if numpy.isnan(stop_i):
+                        stop_i = self.number_of_samples - 1
+                    break
+
+        return start_i, stop_i
+
+    def fixed_threshold_width_forewards(self,
```

Comment:
This method has a typo in "forwards" which may impact people using the library

Reply:
Ekk. Thanks for spotting this!